### ISSUE TITLE
sortFacetValuesBy does not work with integers - facet values are sorted randomly

### ISSUE THREAD

**Issue by daFish** [meilisearch/meilisearch#4559]
**Describe the bug**

I have a field `release_year` in my documents of type `integer`.

```
{
    "objectID": "01be6693-797c-4ec2-a9cb-5618da8d9c88",
    "release_date": "2018-03-23",
    "release_year": 2018,
    "release_decade": 2010
}
```

The field is configured as filterable. Returned facets in `facetDistribution` are **not**  sorted based in the `sortFacetValuesBy`-setting.

**To Reproduce**
1. Add documents like the one above to an index.
2. Configure the index settings to have the desired field to be filterable.
3. Search for documents with configured filters.
5. See wrong sorted facets in `facetDistribution`.

**Expected behavior**
The facets are sorted correctly according to the `sortFacetValuesBy`-setting.

**Meilisearch version:**
Meilisearch 1.7.4

**Comment by Kerollmops** [comment]
Hello @daFish 👋

Thank you for the report. However, I investigated the matter and confirmed that Meilisearch outputs the facet values in 
_alpha_ or _count_ order depending on the `faceting.sortFacetValuesBy` setting. I used the v1.7.6 to do my tests.

However, you are correct that Meilisearch will not try to sort facet values in _alpha_ order if they are **sent as numbers**. You can see in the following code sample the way facets are output when the default (_alpha_) setting is used. It is outputted in a non-particular order.

```json
"facetDistribution": {
    "release_year": {
        "2010": 1,
        "2021": 1,
        "2024": 1,
        "2011": 1,
        "2015": 1,
        "2023": 1
    }
}
```

On the other hand, sorting this `release_year` facet by _count_ sorts the values correctly, as you can see in the following code sample. The values are sorted in the _most-popular-ones-first_ order.

```json
"facetDistribution": {
    "release_year": {
        "2023": 3,
        "2021": 2,
        "2015": 2,
        "2011": 2,
        "2024": 1,
        "2010": 1,
        "2009": 1
    }
}
```

**Comment by daFish** [comment]
Thank you, @Kerollmops, for the detailed answer. Sending the values as strings will fix my issue. However, this fact warrants a note in the documentation, don't you think? 😃

**Comment by Kerollmops** [comment]
> However, this fact warrants a note in the documentation, don't you think? 😃

Yeah, I am first investigating the time it would take to make it work even for numbers. If it is too hard to implement (for now) we'll add it to the documentation 😃

**Comment by Kerollmops** [comment]
Hey @daFish 👋

I fixed your issue in https://github.com/meilisearch/meilisearch/pull/4581. It will be available in v1.8.0 (in 3 weeks) 🎉

**Comment by daFish** [comment]
This is amazing. Thank you @Kerollmops.
